### DISCUSSION TITLE
KMAC key min 4B length check

### DISCUSSION THREAD

**Discussion by manastasova** [openssl/openssl#25833] [Q&A] [answered]
OpenSSL3 adds KMAC support. I noticed there is a length check on the input key K for KMAC: KMAC(K, X, L, S), limiting it to min 4B.

[#define KMAC_MIN_KEY](https://github.com/openssl/openssl/blob/8ff6edb9da6199b130bfb50bc27b2e58cc815932/providers/implementations/macs/kmac_prov.c#L102)
[check on KMAC_MIN_KEY](https://github.com/openssl/openssl/blob/8ff6edb9da6199b130bfb50bc27b2e58cc815932/providers/implementations/macs/kmac_prov.c#L268)

[NIST.SP.800-185](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-185.pdf) defines K as: "_K is a key bit string of any length, including zero_". 

Is there a particular reason for this additional check?

**Comment by paulidale** [comment, marked as answer]
[SP 800-131a rev 2](https://csrc.nist.gov/pubs/sp/800/131/a/r2/final) section 10 specifies a minimum key length of 112 bits.  We allow less for backwards compatibility but not down to zero (which gives you zero security).

**Reply by manastasova** [reply]
Thank you! That is exactly what I needed!
